plugins/out_doris/doris_conf.c

```diff
@@ -0,0 +1,241 @@
+			         cur_bytes/1024/1024, cur_rows, total_speed_mbps, total_speed_rps,
+			         inc_time, inc_speed_mbps, inc_speed_rps);
+        
+        last_time = cur_time;
+		last_bytes = cur_bytes;
+		last_rows = cur_rows;
+    }
+
+    return NULL;
+}
+
+struct flb_out_doris *flb_doris_conf_create(struct flb_output_instance *ins,
+                                            struct flb_config *config)
+{
+    int ret;
+    int io_flags = 0;
+    const char *tmp;
+    struct flb_upstream *upstream;
+    struct flb_out_doris *ctx = NULL;
+    struct flb_doris_progress_reporter *reporter = NULL;
+
+    /* Allocate plugin context */
+    ctx = flb_calloc(1, sizeof(struct flb_out_doris));
+    if (!ctx) {
+        flb_errno();
+        return NULL;
+    }
+    ctx->ins = ins;
+
+    ret = flb_output_config_map_set(ins, (void *) ctx);
+    if (ret == -1) {
+        flb_free(ctx);
+        return NULL;
+    }
+
+    /* Set default network configuration */
+    flb_output_net_default("127.0.0.1", 8030, ins);
+
+    /* Validate */ 
+    if (!ctx->endpoint_type || (strcasecmp(ctx->endpoint_type, "fe") != 0 && strcasecmp(ctx->endpoint_type, "be") != 0)) {
+        flb_plg_error(ins, "endpoint_type is invalid");
+        flb_free(ctx);
+        return NULL;
+    }
+    if (!ctx->user) {
+        flb_plg_error(ins, "user is not set");
+        flb_free(ctx);
+        return NULL;
+    }
+    if (!ctx->database) {
+        flb_plg_error(ins, "database is not set");
+        flb_free(ctx);
+        return NULL;
+    }
+    if (!ctx->table) {
+        flb_plg_error(ins, "table is not set");
+        flb_free(ctx);
+        return NULL;
+    }
+
+    /* Check if SSL/TLS is enabled */
+#ifdef FLB_HAVE_TLS
+    if (ins->use_tls == FLB_TRUE) {
+        io_flags = FLB_IO_TLS;
+    }
+    else {
+        io_flags = FLB_IO_TCP;
+    }
+#else
+    io_flags = FLB_IO_TCP;
+#endif
+
+    if (ins->host.ipv6 == FLB_TRUE) {
+        io_flags |= FLB_IO_IPV6;
+    }
+
+    /* Prepare an upstream handler */
+    upstream = flb_upstream_create(config,
+                                   ins->host.name,
+                                   ins->host.port,
+                                   io_flags, ins->tls);
+
+    if (!upstream) {
+        flb_free(ctx);
+        return NULL;
+    }
+
+    /* url: /api/{database}/{table}/_stream_load */
+    snprintf(ctx->uri, sizeof(ctx->uri), "/api/%s/%s/_stream_load", ctx->database, ctx->table);
+
+    /* label prefix */
+    ctx->add_label = 1;
+    tmp = flb_output_get_property("label_prefix", ins);
+    if (tmp) {
+        /* Just check if we have to disable it */
+        if (flb_utils_bool(tmp) == FLB_FALSE) {
+            ctx->add_label = 0;
+        }
+    }
+
+    /* Output format */
+    ctx->out_format = FLB_PACK_JSON_FORMAT_LINES;
+
+    /* Date key */
+    ctx->date_key = ctx->time_key;
+    tmp = flb_output_get_property("time_key", ins);
```

Comment:
Is this meant to allow the user to select any member but `off` and `false` as fields to pull hte date from?
If that's the case then it's ok but it should be clearly stated in the documentation to avoid issues.

On the other hand, the usage of `flb_output_get_property` needs to be removed as you are mixing old and new features which will cause issues in the future.